iac/mainnet-fork/terraform/main.tf

```diff
@@ -0,0 +1,276 @@
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "3.74.2"
+    }
+  }
+}
+
+data "terraform_remote_state" "setup_iac" {
+  backend = "s3"
+  config = {
+    bucket = "aztec-terraform"
+    key    = "setup/setup-iac"
+    region = "eu-west-2"
+  }
+}
+
+data "terraform_remote_state" "aztec-network_iac" {
+  backend = "s3"
+  config = {
+    bucket = "aztec-terraform"
+    key    = "aztec-network/iac"
+    region = "eu-west-2"
+  }
+}
+
+data "terraform_remote_state" "aztec2_iac" {
+  backend = "s3"
+  config = {
+    bucket = "aztec-terraform"
+    key    = "aztec2/iac"
+    region = "eu-west-2"
+  }
+}
+
+
+data "aws_alb" "aztec-network_alb" {
+  arn = data.terraform_remote_state.aztec2_iac.outputs.alb_arn
+}
+
+provider "aws" {
+  profile = "default"
+  region  = "eu-west-2"
+}
+
+resource "aws_service_discovery_service" "aztec_mainnet_fork" {
+  name = "aztec-network-mainnet-fork"
+
+  health_check_custom_config {
+    failure_threshold = 1
+  }
+
+  dns_config {
+    namespace_id = data.terraform_remote_state.setup_iac.outputs.local_service_discovery_id
+
+    dns_records {
+      ttl  = 60
+      type = "A"
+    }
+
+    dns_records {
+      ttl  = 60
+      type = "SRV"
+    }
+
+    routing_policy = "MULTIVALUE"
+  }
+}
+
+# EFS filesystem for mainnet fork
+resource "aws_efs_file_system" "aztec_mainnet_fork_data_store" {
+  creation_token = "aztec-network-mainnet-fork-data"
+
+  tags = {
+    Name = "aztec-network-mainnet-fork-data"
+  }
+
+  lifecycle_policy {
+    transition_to_ia = "AFTER_30_DAYS"
+  }
+}
+
+resource "aws_efs_mount_target" "aztec_fork_private_az1" {
+  file_system_id  = aws_efs_file_system.aztec_mainnet_fork_data_store.id
+  subnet_id       = data.terraform_remote_state.setup_iac.outputs.subnet_az1_private_id
+  security_groups = [data.terraform_remote_state.setup_iac.outputs.security_group_private_id]
+}
+
+resource "aws_efs_mount_target" "aztec_fork_private_az2" {
+  file_system_id  = aws_efs_file_system.aztec_mainnet_fork_data_store.id
+  subnet_id       = data.terraform_remote_state.setup_iac.outputs.subnet_az2_private_id
+  security_groups = [data.terraform_remote_state.setup_iac.outputs.security_group_private_id]
+}
+
+# Define deployment task and service
+resource "aws_ecs_task_definition" "aztec_mainnet_fork" {
+  family                   = "aztec-network-mainnet-fork"
+  requires_compatibilities = ["FARGATE"]
+  network_mode             = "awsvpc"
+  cpu                      = "2048"
+  memory                   = "4096"
+  execution_role_arn       = data.terraform_remote_state.setup_iac.outputs.ecs_task_execution_role_arn
+
+  volume {
+    name = "efs-data-store"
+    efs_volume_configuration {
+      file_system_id = aws_efs_file_system.aztec_mainnet_fork_data_store.id
+    }
+  }
+
+  container_definitions = <<DEFINITION
+[
+  {
+    "name": "aztec-network-mainnet-fork",
+    "image": "278380418400.dkr.ecr.eu-west-2.amazonaws.com/aztec-mainnet-fork:aztec3-packages-prod",
```

Comment:
As the repo is public I have been using an env var for the ECR URL.

Reply:
right, makes sense but we do also have this in all our Dockerfiles 😬